### DISCUSSION TITLE
Policy to charge house battery first 2xFronius (Symo + Gen24) + BYD

### DISCUSSION THREAD

**Discussion by Alex0176** [ngardiner/TWCManager#463] [Ideas]
Hallo, 

it will be great to have a policy or flag to charge the house battery first and not the car(s).

**Comment by MikeBishop** [comment]
If your house battery is a Powerwall, it's easy to add a restriction to your Track Green Energy policy to specify a certain battery level.  For example:

      "restrictions": {
        "Track Green Energy": {
          "match": [
            "modules.TeslaPowerwall2.batteryLevel"
          ],
          "condition": [
            "gte"
          ],
          "value": [
            "modules.TeslaPowerwall2.reservePercent"
          ]
        }
      },

**Comment by Alex0176** [comment]
i have a 2 fronius and on one (gen 24) its a byd battery connected. The fornius offer the battery level in % and i like to load the battery before the car.

**Comment by MikeBishop** [comment]
I don't know whether the Fronius exposes the battery percentage as a module property.  If it does, there should be an equivalent restriction.; if it doesn't, that's a more concrete feature request.

**Reply by Alex0176** [reply]
where can i find the propertys?

**Reply by Alex0176** [reply]
i find the SOC in the Fronius API 

/solar_api/v1/GetPowerFlowRealtimeData.fcgi?Scope=System

```
{
   "Body" : {
      "Data" : {
         "Inverters" : {
            "1" : {
               "Battery_Mode" : "normal",
               "DT" : 1,
               "E_Day" : null,
               "E_Total" : 7752168.5352777774,
               "E_Year" : null,
               "P" : -716.2047119140625,
               "**SOC**" : 10.5
            }
         },
         "Site" : {
            "BackupMode" : false,
            "BatteryStandby" : true,
            "E_Day" : null,
            "E_Total" : 7752168.5352777774,
            "E_Year" : null,
            "Meter_Location" : "grid",
            "Mode" : "bidirectional",
            "P_Akku" : -1700.95263671875,
            "P_Grid" : -91.0,
            "P_Load" : 749.3585205078125,
            "P_PV" : 1088.8841552734375,
            "rel_Autonomy" : 100.0,
            "rel_SelfConsumption" : 100.0
         },
         "Smartloads" : {
            "Ohmpilots" : {}
         },
         "Version" : "12"
      }
   },
   "Head" : {
      "RequestArguments" : {},
      "Status" : {
         "Code" : 0,
         "Reason" : "",
         "UserMessage" : ""
      },
      "Timestamp" : "2022-08-25T15:39:16+00:00"
   }
}
```

But this value was not used in the Fronius Module.

**Reply by Alex0176** [reply]
> I don't know whether the Fronius exposes the battery percentage as a module property. If it does, there should be an equivalent restriction.; if it doesn't, that's a more concrete feature request.

Hi Mike, can you check my code and if its ok integrate it into the project? 
best regards
Alex

**Comment by Alex0176** [comment]
Modification:
[Fronius.zip](https://github.com/ngardiner/TWCManager/files/9460528/Fronius.zip)

```
class Fronius:
    soc = -1


```

```
get Value function

    def getSOC(self):

        if not self.status:
            logger.debug("Fronius EMS Module Disabled. Skipping getSOC")
            return 0

        # Perform updates if necessary
        self.update()

        # Return consumption value (negated, as this is how it's presented)
        return float(self.soc)


```

parse the value from the json file



```
                   try:
                        for Inverter in meterData["Body"]["Data"]["Inverters"]:
                            if "SOC" in Inverter:
				    if float(Inverter["SOC"])> soc
                            coc = float(Inverter["SOC"])
                    except (KeyError, TypeError) as e:
                        logger.log(
                            logging.INFO4,
                            "Exception during parsing Meter Data (State of Charge)",
                        )
                        logger.debug(e)
            # Update values
            self.consumedW = con
            self.generatedW = gen
            self.soc = soc

```






i am not a python programmer, but i think this should do the work with getting the value from the inverter, but anyone with phyton knowledge should look over it and told me how i can make the rule (maybe commit it into the project) 

is this in my project anything like this? 

```
  "restrictions": {
    "Track Green Energy": {
      "match": [
        "modules.Fronius.getSOC"
      ],
      "condition": [
        "gte"
      ],
      "value": [
        "99.0"
      ]
    }
  },
```
